classes/checkout/CheckoutPaymentStep.php

```diff
@@ -50,11 +55,13 @@ public function __construct(
         Context $context,
         TranslatorInterface $translator,
         PaymentOptionsFinder $paymentOptionsFinder,
-        ConditionsToApproveFinder $conditionsToApproveFinder
+        ConditionsToApproveFinder $conditionsToApproveFinder,
+        bool $isMultiShipmentFeatureFlagEnabled,
```

Comment:
```suggestion
        bool $isMultiShipmentFeatureFlagEnabled = false,
```

to prevent BC breaks, imho this has to be an optional param